### PR TITLE
Docs: Configure SAML SLO properly with NameID and SessionIndex support

### PR DESCRIPTION
**Pull request by mgyongyosi**
**What is this feature?**
Adding docs on how to configure SAML SLO correctly with the latest Grafana improvements to external session handling.

**Why do we need this feature?**

**Who is this feature for?**

[Add information on what kind of user the feature is for.]

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

### CODE REVIEW

**Review by irenerl24 — APPROVED** [review]

File: docs/sources/setup-grafana/configure-security/configure-authentication/saml/index.md [hunk 1]
```diff
@@ -373,6 +373,20 @@ SAML's single logout feature allows users to log out from all applications assoc
 `HTTP-Redirect` and `HTTP-POST` bindings are supported for single logout.
 When using `HTTP-Redirect` bindings the query should include a request signature.
 
+#### Configure single logout
+
+{{% admonition type="note" %}}
```
Comment (vtorosyan) [inline docs/sources/setup-grafana/configure-security/configure-authentication/saml/index.md:378]: I always find those sections starting with "disclaimers/notes" less readable to be honest, as the first I want to see when I read "Configure single logout" is to learn how to configure single logout. In other words, I'd suggest to move the note down the next paragraph.

File: docs/sources/setup-grafana/configure-security/configure-authentication/saml/index.md [hunk 2]
```diff
@@ -373,6 +373,20 @@ SAML's single logout feature allows users to log out from all applications assoc
...
+Proper `NameID` and `SessionIndex` support is available in public preview behind the `improvedExternalSessionHandling` feature toggle, starting from Grafana v11.5.
```
Comment (vtorosyan) [inline docs/sources/setup-grafana/configure-security/configure-authentication/saml/index.md:379]: It is a bit confusing to read "proper `NameID", as it is unclear what is proper. I'd probably suggest first to have a small section on it's own describing the changes with `NameID` and `SessionIndex`, even before the `Configure single logout` section. And once this is done, we can simply change this note to say something like:

```
{{% admonition type="note" %}}
The improvedExternalSessionHandling feature toggle is available in public preview starting from Grafana v11.5. To enable it, refer to Feature Toggles Documentation.
{{% /admonition %}}
```

The section above this, can be something like

```
#### `NameID` and `SessionIndex` changes in Grafana v11.5

Before Grafana v11.5:
- Grafana used the Login attribute (from `assertion_attribute_login`) as the `NameID` in logout requests. This often caused issues if the `Login` value did not match the IdP’s expectations.
- Grafana did not support the `SessionIndex`, leading to potential issues like logging out all IdP sessions when logging out of Grafana.

Starting from Grafana v11.5
- Grafana uses the NameID from the SAML assertion for logout requests. If NameID is missing, Grafana defaults to the Login attribute.
- Grafana now supports including the SessionIndex in logout requests if it is provided by the IdP.
```

Does not have to be this exact format, but I hope I could explain the idea.

File: docs/sources/setup-grafana/configure-security/configure-authentication/saml/index.md [hunk 3]
```diff
@@ -373,6 +373,20 @@ SAML's single logout feature allows users to log out from all applications assoc
...
+{{% /admonition %}}
+
+Before Grafana version 11.5, the `Login` attribute value (extracted from the SAML assertion using the `assertion_attribute_login` configuration) was used as the `NameID` in the logout request. This could cause issues with single logout if the `assertion_attribute_login` value differed from what the Identity Provider (IdP) expected.
```
Comment (vtorosyan) [inline docs/sources/setup-grafana/configure-security/configure-authentication/saml/index.md:382]: If you decide to do the above mentioned suggestions, the `Configure single logout` section can be edited to something like:

```
To configure single logout in Grafana:
	1.	Enable the `single_logout option` in your configuration.
	2.	Ensure the `name_id_format` matches the format your IdP expects (e.g., `urn:oasis:names:tc:SAML:2.0:nameid-format:transient`).
	3.	Enable the `improvedExternalSessionHandling` feature toggle for full NameID and SessionIndex support (Grafana v11.5+).
```

**Review by vtorosyan — APPROVED** [review]
I am approving the PR as the content looks good to me. I left few suggestions about the structure and if you decide to adapt the suggestions, feel free to merge the PR afterwards or request review again. 

Thanks for the PR!

**Pull request merged**